devicetypes/plaidsystems/spruce-controller.src/spruce-controller.groovy

```diff
@@ -14,428 +10,368 @@
-        attribute 'switch16', 'string'
-		attribute 'rainsensor', 'string'
-        attribute 'status', 'string'
-        attribute 'tileMessage', 'string'
-        attribute 'minutes', 'string'
-        attribute 'VALUE_UP', 'string'
-        attribute 'VALUE_DOWN', 'string'        
-        
-        command 'levelUp'
-        command 'levelDown'
+	definition (name: "Spruce Controller", namespace: "plaidsystems", author: "plaidsystems", mnmn: "SmartThingsCommunity", vid: "412ceebe-09e5-3b06-8ed6-ab12ec248cc7"){
+		capability "Actuator"
+        capability "Switch"        
+        capability "Water Sensor"
+        capability "Sensor"
+        capability "Health Check"
+        capability "heartreturn55003.status"
+        capability "heartreturn55003.controllerMessage"
+        capability "heartreturn55003.rainSensor"
+        
+        capability "Configuration"
+        capability "Refresh"        
+        
+        attribute "status", "string"
+        attribute "controllerMessage", "string"
+        attribute "rainsensor", "string"
+        
+        command "on"
+        command "off"        
+        command "valveOn"
+        command "valveOff"
+        command "zoneDuration"
+        
+        command "wet"
+        command "dry"
+        
+        command "setStatus"
+        command "setRainsensor"
+        command "setControllerMessage"
+        
+        command "zon"
+        command "zoff"
         command 'programOn'
         command 'programOff'
         command 'programWait'
         command 'programEnd'
         
-        command 'on'
-        command 'off'
-        command 'zon'
-        command 'zoff'
-        command 'z1on'
-		command 'z1off'
-		command 'z2on'
-		command 'z2off'        
-		command 'z3on'
-		command 'z3off'
-		command 'z4on'
-		command 'z4off'
-		command 'z5on'
-		command 'z5off'
-		command 'z6on'
-		command 'z6off'
-		command 'z7on'
-		command 'z7off'
-		command 'z8on'
-		command 'z8off'
-        command 'z9on'
-        command 'z9off'
-		command 'z10on'
-		command 'z10off'
-		command 'z11on'
-		command 'z11off'
-		command 'z12on'
-		command 'z12off'
-		command 'z13on'
-		command 'z13off'
-		command 'z14on'
-		command 'z14off'
-		command 'z15on'
-		command 'z15off'
-		command 'z16on'
-		command 'z16off'
-        
-        command 'config'
-        command 'refresh'        
-        command 'rain'
-        command 'manual'
-        command 'manualTime'
-        command 'settingsMap'
-        command 'writeTime'
-        command 'writeType'        
-        command 'notify'
-        command 'updated'      
-        
-		//ST release
-		//fingerprint endpointId: '1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18', profileId: '0104', deviceId: '0002', deviceVersion: '00', inClusters: '0000,0003,0004,0005,0006,000F', outClusters: '0003, 0019', manufacturer: 'PLAID SYSTEMS', model: 'PS-SPRZ16-01'
+        command "config"
+        command "refresh"        
+        command "rain"
+        command "settingsMap"
```

Comment:
Did you defined these commands in your custom capabilities?

Reply:
Ok, removed unused commands.  Commands listed are custom capability or the standard "switch" commands